icefoganalytics/sfa-client
# Messaging Module: Template Updates: YG Rejection Letter

Fixes https://github.com/icefoganalytics/sfa-client/issues/48

# Context

Translate "YG Rejection Letter.docx" into an html/pdf template, and serve it up via the appropriate endpoint.

## Questions/Considerations

1. Should the http://localhost:3000/api/v2/admin/funding-requests/265/letters endpoint only return the letters that are appropriate for the given funding request status? Or should it do what its currently doing, i.e. returning all possible letters for the funding request type.
2. Should I update the `www.yukonstudentaid.com` link to the link it resolves to? i.e. https://yukon.ca/en/education-and-schools-student-financial-support
3. Can a funding request be rejected for multiple rejection reasons? According to the database a funding request can only have a single status reason but the wording of this letter suggests that multiple reasons are possible.
4. Do the rejection reasons need a UI translation? I generated one letter and it looked like this, which seems a little off to me.
![image](https://github.com/icefoganalytics/sfa-client/assets/23045206/fa22f453-8871-40a3-af88-de794f8510bf)


# Testing Instructions

1. Boot the api and database services via `dev up`
6. Boot the web service via `cd src/web && npm run start`
7. Log in to the app at http://localhost:8080
8. Go to http://localhost:3000/api/v2/admin/funding-request-letters and see that there is now a rejection letter for the Yukon Grant.
9. Go to http://localhost:3000/api/v2/admin/funding-requests/265/letters and see that there are now multiple letter options for the specific letter.
10. Go to http://localhost:3000/api/v2/admin/funding-requests/265/letters/student.json and see that the template data is generated correctly. Also note that because Funding Request 256 was rejected, the template that will be used is a rejection template.
11. Go to http://localhost:3000/api/v2/admin/funding-requests/265/letters/student.html and check that the general Yukon Grant Rejection letter template looks correct.
12. Go to http://localhost:3000/api/v2/admin/funding-requests/265/letters/student.pdf and check that the letter pdf looks correct.
